### PR TITLE
Wrap "yt" package

### PR DESCRIPTION
**Pull request by eschnett**
yt is a python package for analyzing and visualizing volumetric, multi-resolution data from astrophysical simulations, radio telescopes, and a burgeoning interdisciplinary community.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    depends_on("mercurial", type="build")
+    depends_on("py-astropy", when="+astropy")
+    depends_on("py-cython")
+    depends_on("py-h5py", when="+h5py")
+    depends_on("py-ipython")
+    depends_on("py-matplotlib")
+    depends_on("py-numpy")
+    depends_on("py-scipy", when="+scipy")
+    depends_on("py-setuptools", type="build")
+    depends_on("py-sympy")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Most python module dependencies are `type=nolink`, meaning they are needed to build and run, but not for linking.

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    version("development", hg="https://bitbucket.org/yt_analysis/yt",
+            branch="yt")
+
+    variant("astropy", default=True, description="enable astropy support")
+    variant("h5py", default=True, description="enable h5py support")
+    variant("scipy", default=True, description="enable scipy support")
+
+    extends("python")
+
+    depends_on("mercurial", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: I mentioned this in #1750, but I don't think this package should depend on mercurial any more than packages that download from git should depend on git, or packages that download through curl should depend on curl.

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    def install(self, spec, prefix):
+        # python("setup.py", "install", "--user", "--prefix=%s" % prefix)
+        python("setup.py", "install", "--prefix=%s" % prefix)
+        self.check_install(spec)
+
+    def check_install(self, spec):
+        # The Python interpreter path can be too long for this
+        # yt = Executable(join_path(prefix.bin, "yt"))
+        # yt("--help")
+        python = which("python")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: This line shouldn't be necessary.

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    depends_on("py-scipy", when="+scipy")
+    depends_on("py-setuptools", type="build")
+    depends_on("py-sympy")
+    depends_on("python @2.7:2.999,3.4:")
+
+    # TODO: add py-ipython
+
+    def install(self, spec, prefix):
+        # python("setup.py", "install", "--user", "--prefix=%s" % prefix)
+        python("setup.py", "install", "--prefix=%s" % prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Can you replace `python("setup.py", ` with `setup_py(`? See #950 and #1424

File: var/spack/repos/builtin/packages/py-yt/package.py (hunk 2 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Mercurial is not always installed. My hope is that this dependency (maybe with `type="fetch"`?) would make things work on all systems, without people having to manually install Mercurial first.

On the other hand, Spack should just do this automatically.

Or we ask the `yt` developers to provide tarballs, as most other projects do.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Yes, but by the same logic, shouldn't every package that downloads from git depend on git? And every package that downloads with curl depend on curl? And every package that needs a compiler depends on a compiler? There are certain things like this that are more dependencies of Spack than dependencies of a package.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Ideally, Spack would check your path and see if git or curl or mercurial are installed, and if they aren't, then install them. But I don't think we should start adding them as build dependencies of every package.

File: var/spack/repos/builtin/packages/py-yt/package.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    depends_on("py-astropy", type="nolink", when="+astropy")
+    depends_on("py-cython", type="nolink")
+    depends_on("py-h5py", type="nolink", when="+h5py")
+    depends_on("py-ipython", type="nolink")
+    depends_on("py-matplotlib", type="nolink")
+    depends_on("py-numpy", type="nolink")
+    depends_on("py-scipy", type="nolink", when="+scipy")
+    depends_on("py-setuptools", type="build")
+    depends_on("py-sympy", type="nolink")
+    depends_on("python @2.7:2.999,3.4:", type="nolink")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-yt/package.py]: I don't think Python should be `type=nolink`. At least, I've never seen any other packages do that. But the Python modules look better now, thanks!

Comment (eschnett) [inline var/spack/repos/builtin/packages/py-yt/package.py]: Corrected.

**Comment by davydden** [comment]
you may want to squash your commits into a single one to keep the history clean.

**Comment by adamjstewart** [comment]
LGTM, thanks @eschnett

**Comment by eschnett** [comment]
Squashed.

**Comment by tgamblin** [comment]
No need to squash in the PR -- I can do that on merge. Thanks!

**Pull request merged**